### PR TITLE
remove matplotlib from requirements, and making it an optional dep. 

### PR DESCRIPTION
**Pull request by CamDavidsonPilon**
This reduces the dep if you want a cleaner stack. Interestingly enough, Pandas doesn't have mpl as a dependency.

### CODE REVIEW

**Comment by CamDavidsonPilon** [comment]
@spacecowboy any thoughts on this?

File: lifelines/plotting.py [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
-# plotting
 import numpy as np
-import matplotlib as mpl
-from matplotlib import pyplot as plt
-
 from lifelines.utils import coalesce
 
 
 def is_latex_enabled():
+    import matplotlib as mpl
```
Comment (spacecowboy) [inline lifelines/plotting.py]: Docstrings have to be the *first* thing in the method, no?
So it should be moved to after the docstring.

**Comment by spacecowboy** [comment]
Seems reasonable otherwise

File: lifelines/plotting.py (hunk 1 above)
Comment (CamDavidsonPilon) [inline lifelines/plotting.py]: :+1:

**Pull request merged**